sync/notify/__init__.py

```diff
@@ -0,0 +1,44 @@
+import geckomsg
+import wptfyimsg
+
+
+def message_for_sync(sync):
+    wptfyi_parts = wptfyimsg.for_sync(sync)
+    gecko_parts = geckomsg.for_sync(sync)
+
+    msg_parts = []
+    if wptfyi_parts is not None:
+        msg_parts.extend(wptfyi_parts)
+    if gecko_parts is not None:
+        msg_parts.extend(gecko_parts)
+
+    if not msg_parts:
+        return
+
+    truncated, truncated_message = truncate_message(msg_parts)
+    if truncated:
+        message = "\n".join(msg_parts)
+    else:
+        message = truncated_message
+        truncated_message = None
+
+    return message, truncated_message
+
+
+def truncate_message(parts):
+    # This is currently the number of codepoints, with the proviso that
+    # only BMP characters are supported
+    MAX_LENGTH = 65535
+
+    suffix = "\n(truncated for maximum comment length)"
+    message = ""
+    for part in parts:
+        if len(message) + len(part) + 1 > MAX_LENGTH - len(suffix):
+            message += suffix
```

Comment:
If the first part is over the max length, then the message will only contain the truncated notification and nothing else. Should we not max out the message with content if possible?

Reply:
In practice that can't happen because the first part is never long enough. But I'll change the truncation text to mention the attachment so that it's useful even in that unlikley case.